pkg/operator2/configobservation/infrastructure/observe_apiserverurl.go

```diff
@@ -0,0 +1,52 @@
+package infrastructure
+
+import (
+	"fmt"
+	"net/url"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+)
+
+func ObserveAPIServerURL(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, _ []error) {
```

Comment:
it looks close to `ObserveConsoleURL` - only `source` and `destination` are different but i'm not sure if we want to factor it out.

Reply:
The listers and the resources you're grabbing the information from are different, too. I think we might end up with a lot of arguments and abstraction just to factor it out, I would say probably not worth it.